### PR TITLE
Fix resolve_type for containers of generic types [GRF-6059]

### PR DESCRIPTION
**Pull request by bcalvert-graft**
# Description
The function added in #7 was missing some corner cases: in particular, it needs to be able to handle containers of generic types. This PR fixes it.
## Testing
```bash
$ pytest \
    sematic/utils/tests/test_types.py \
    sematic/types/types/tests/test_dataclass_generic.py
```

### CODE REVIEW

File: sematic/utils/types.py [hunk 1]
```diff
@@ -67,20 +72,38 @@ def resolve_type(cls: type, attribute: str):
...
+) -> Union[type, TypeVar]:
+    origin = get_origin(type_)
+    if origin is not None:  # It's a generic like list, dict, etc.
+        resolved_args = tuple(
+            _resolve_generic_type(cls=cls, type_=arg, attribute=attribute)
+            for arg in get_args(type_)
+        )
+        major, minor, *_ = sys.version_info
+        if (major, minor) >= (3, 10) and issubclass(origin, types.UnionType):
+            return cast(type, Union[resolved_args])
```
Comment (eugene-graft) [inline sematic/utils/types.py:89]: Nice!

**Review by eugene-graft — APPROVED** [review]
LGTM
